packages/core/src/domain/error/error.ts

```diff
@@ -31,11 +31,12 @@ export function computeRawError({
 }: RawErrorParams): RawError {
   const isErrorInstance = originalError instanceof Error
 
-  const message = stackTrace?.message
-    ? stackTrace.message
-    : !isErrorInstance
-    ? `${nonErrorPrefix} ${jsonStringify(sanitize(originalError))!}`
-    : 'Empty message'
+  const message =
+    stackTrace?.message && stackTrace?.name
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I don't understand why we check for `stackTrace?.name` here. If this is now a required condition to consider whether we should take `message` into account, we could handle this when creating the `StackTrace` object and don't set a `stackTrace.message` if there is no `name`. WDYT?

Reply:
> I don't understand why we check for `stackTrace?.name` here

The idea is that if we did not manage to get a name, it means that we are in a case where the message would not be great either and we already generate our own message for those ones. We need to add this extra check with the new path taken by non error uncaught exception.

> we could handle this when creating the StackTrace object and don't set a stackTrace.message if there is no name. WDYT?

For now, I tried to not impact tracekit logic/design that much but we could reduce several complexity spots reworking some parts and by making it more integrated with our own error logic.
I don't think that is the right time to do it though, I can collect some thoughts about that in a card for future us.
wdyt?

Reply:
Okay. You could add a comment to give some context here, and maybe add a TODO to explain that we should rework tracekit design in the future.

Reply:
comments added in https://github.com/DataDog/browser-sdk/pull/2182/commits/48e7bdff517a641f31699c43de4273a548c4bad1